### ISSUE TITLE
[Feature] Add email worker support

### ISSUE THREAD

**Issue by zengboin** [cloudflare/workers-rs#274]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

Please add email event/worker support

**Comment by benthecarman** [comment]
+1 this would be great to have

**Comment by GlenDC** [comment]
I would be a first time contributor, but if someone can point me to some kind of semi-structured plan and some pointers I wouldn't mind contributing support for this myself. Happy to help like that!

**Comment by GenshinMinecraft** [comment]
+1, But this Issue has been on the table for 2 years now, and I don't see it happening anytime soon.

**Comment by spigaz** [comment]
There is a PR open already for it, I'm not sure why @LuisDuarte1 hasn't pushed the requested changes.

You could either talk with him or pick up where he where he left of.

**Comment by LuisDuarte1** [comment]
> You could either talk with him or pick up where he where he left of.

I probably only be to able to pick it up in 3 weeks time, lots of stuff going on right now 😅

Feel free to take my work if you want to, the only think not working was that bindings weren't being set properly, for some reason

**Comment by NiiightmareXD** [comment]
Are there any updates?

**Comment by spigaz** [comment]
I'm guessing this native api still uses mailchannels.
https://blog.cloudflare.com/sending-email-from-workers-with-mailchannels/

There are alternatives, for instance, postmark or resend with a post request, resent has a crate but it seems its not working.
https://developers.cloudflare.com/workers/tutorials/send-emails-with-postmark/
https://developers.cloudflare.com/workers/tutorials/send-emails-with-resend/

If you still want to use the native using mailchannels, you have to register with them:
https://support.mailchannels.com/hc/en-us/articles/4565898358413-Sending-Email-from-Cloudflare-Workers-using-MailChannels-Email-API

Comments from support with some alternatives:
https://community.cloudflare.com/t/mailchannels-end-of-life-notice-migrate-now/658638

But I guess you can use any provider that accepts a post request to send an email or has a working crate.

**Comment by NiiightmareXD** [comment]
> I'm guessing this native api still uses mailchannels. https://blog.cloudflare.com/sending-email-from-workers-with-mailchannels/
> 
> There are alternatives, for instance, postmark or resend with a post request, resent has a crate but it seems its not working. https://developers.cloudflare.com/workers/tutorials/send-emails-with-postmark/ https://developers.cloudflare.com/workers/tutorials/send-emails-with-resend/
> 
> If you still want to use the native using mailchannels, you have to register with them: https://support.mailchannels.com/hc/en-us/articles/4565898358413-Sending-Email-from-Cloudflare-Workers-using-MailChannels-Email-API
> 
> Comments from support with some alternatives: https://community.cloudflare.com/t/mailchannels-end-of-life-notice-migrate-now/658638
> 
> But I guess you can use any provider that accepts a post request to send an email or has a working crate.

How about receiving emails?

**Comment by spigaz** [comment]
Yes, you are correct, you were talking about email workers, my mistake:
https://developers.cloudflare.com/email-routing/email-workers/

Well there is the PR that @LuisDuarte1 did, but he wasn't able to pick up since
https://github.com/cloudflare/workers-rs/pull/624

I'm not sure if its usable, but it might be, he even did an example, I would try it if I were you...
